fusor-ember-cli/app/routes/deployment.js

```diff
@@ -116,32 +116,52 @@ export default Ember.Route.extend(DeploymentRouteMixin, {
       controller.set('showSpinner', true);
 
       request({
-          url: '/fusor/api/v21/deployments/' + deployment.get('id') + '/deploy' ,
-          type: "PUT",
-          headers: {
-              "Accept": "application/json",
-              "Content-Type": "application/json",
-              "X-CSRF-Token": token,
-              "Authorization": "Basic " + self.get('session.basicAuthToken')
-          }
-          }).then(function(response) {
-            var uuid = response.id;
-            console.log('task uuid is ' + uuid);
-            deployment.set('foreman_task_uuid', uuid);
-            deployment.save().then(function () {
+        url: '/fusor/api/v21/deployments/' + deployment.get('id') + '/deploy',
+        type: "PUT",
+        headers: {
+          "Accept": "application/json",
+          "Content-Type": "application/json",
+          "X-CSRF-Token": token,
+          "Authorization": "Basic " + self.get('session.basicAuthToken')
+        }
+      }).then(
+        function (response) {
+          var uuid = response.id;
+          console.log('task uuid is ' + uuid);
+          deployment.set('foreman_task_uuid', uuid);
+          deployment.save().then(
+            function () {
+              controller.set('showSpinner', false);
               return self.transitionTo('review.progress.overview');
-            }, function () {
+            },
+            function () {
+              controller.set('showSpinner', false);
               controller.set('errorMsg', 'Error in saving UUID of deployment task.');
               controller.set('showErrorMessage', true);
             });
-          }, function(error) {
-            controller.set('showSpinner', false);
-            console.log(error);
-            var errorMsg = error.responseText;
-            controller.set('errorMsg', errorMsg);
+        },
+        function (response) {
+          controller.set('showSpinner', false);
+
+          if (response.jqXHR.status !== 422 || !response.jqXHR.responseJSON || !response.jqXHR.responseJSON.errors) {
```

Comment:
Is this 422 check needed since you removed the possibility of 422 being returned?